salesforce/10.10/modules/ROOT/pages/salesforce-connector-processing-events.adoc

```diff
@@ -9,88 +9,90 @@ You can use Anypoint Connector for Salesforce (Salesforce Connector) as an inbou
-* When running the application in CloudHub, if you are using multiple workers, all of the workers receive and process the events.
+The way that the object store stores the necessary data for the *Resume from the Last Replay Id* varies depending on the environment:
 
-=== Multiple Flows That Use the Same Source
-
-It is best that you do not have two or more flows using a source that listens to events on the same channel or topic because the events will be consumed twice from your user quota. If you have a use case that requires processing the same events in two or more different flows, consider implementing that logic in a single flow instead.
+* When a Mule app runs locally, the connector creates a persistent store in the file system.
++
+If the application stops and restarts, or if a connection issue triggers a reconnection, the data saved in the object store enables the connector to continue processing messages at the point where it previously stopped.
+* When a Mule app runs in a cluster with the nodes either on the same machine or on different machines, Mule uses a distributed in-memory object store:
+** If the primary node (the one consuming the messages) shuts down, the node that takes over for the primary node does not reprocess the same messages. Instead, the node continues from where the previous node stopped processing the messages.
+** If the cluster stops, the data from the object store is lost. When the cluster restarts, the configuration at the source level determines whether the connector reprocesses the messages.
 
-Having multiple sources listening to the same events and having the *Resume from the Last Replay Id* option enabled leads to data corruption and potential loss of events.
++
+To persist data so that the app continues processing events when restarted, create a persistent object store, as described in xref:mule-runtime::creating-and-managing-a-cluster-manually.adoc[Create and Manage a Cluster Manually].
+* If you use multiple workers when running the app in CloudHub, all of the workers receive and process the events.
 
-=== Streaming Channel with No Listener
+== Multiple Flows That Use the Same Source
 
-Users can push events to a streaming channel, even if the channel does not have a listener for reading published events. Once a listener is started for the channel, Salesforce Streaming API pushes as many messages it can to the listener, based on the maximum daily limit.
+Avoid having two or more flows use a source that listens to events on the same channel or topic, because the events will be consumed twice from your user quota. If you have a use case that requires processing the same events in two or more different flows, consider implementing that logic in a single flow instead.
```

Comment:
Just a couple alternatives for your consideration:
```suggestion
If you have two or more flows use a source that listens to events on the same channel or topic, the events are consumed twice from your user quota. To avoid this, consider implementing that logic in a single flow instead.
```
or
```suggestion
To avoid events being consumed twice when you have two or more flows use a source that listens to events on the same channel or topic, consider implementing that logic in a single flow instead.
```